src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -906,18 +908,20 @@ public JSONObject update(JSONObject in) {
             MESSAGE_HANDLE_EDITOR_CLOSED,
             MESSAGE_VARIANTS_RECEIVED
     })
+
+    @Retention(RetentionPolicy.SOURCE)
     public @interface MessageType {}
 
-    private static final int MESSAGE_INITIALIZE_CHANGES = 0;
-    private static final int MESSAGE_CONNECT_TO_EDITOR = 1;
-    private static final int MESSAGE_SEND_STATE_FOR_EDITING = 2;
-    private static final int MESSAGE_HANDLE_EDITOR_CHANGES_RECEIVED = 3;
-    private static final int MESSAGE_SEND_DEVICE_INFO = 4;
-    private static final int MESSAGE_EVENT_BINDINGS_RECEIVED = 6;
-    private static final int MESSAGE_HANDLE_EDITOR_BINDINGS_RECEIVED = 8;
-    private static final int MESSAGE_SEND_EVENT_TRACKED = 9;
-    private static final int MESSAGE_HANDLE_EDITOR_CLOSED = 10;
-    private static final int MESSAGE_VARIANTS_RECEIVED = 11;
+    private static final @MessageType int MESSAGE_INITIALIZE_CHANGES = 0;
+    private static final @MessageType int MESSAGE_CONNECT_TO_EDITOR = 1;
+    private static final @MessageType int MESSAGE_SEND_STATE_FOR_EDITING = 2;
+    private static final @MessageType int MESSAGE_HANDLE_EDITOR_CHANGES_RECEIVED = 3;
+    private static final @MessageType int MESSAGE_SEND_DEVICE_INFO = 4;
+    private static final @MessageType int MESSAGE_EVENT_BINDINGS_RECEIVED = 6;
```

Comment:
I think you're right! I've been reluctant to change these constants (and there were some constants that have gone away) but these should *never* be persisted. I'll make that change!